sdk/storage/azure-storage-file-share/tests/test_file_async.py

```diff
@@ -975,6 +975,47 @@ async def test_list_ranges_none_with_invalid_lease_fails_async(self, resource_gr
         self.assertIsNotNone(ranges)
         self.assertEqual(len(ranges), 0)
 
+    @GlobalStorageAccountPreparer()
+    @AsyncStorageTestCase.await_prepared_test
+    async def test_list_ranges_diff(self, resource_group, location, storage_account, storage_account_key):
+        self._setup(storage_account, storage_account_key)
+        file_name = self._get_file_reference()
+        await self._setup_share(storage_account, storage_account_key)
+        file_client = ShareFileClient(
+            self.account_url(storage_account, "file"),
+            share_name=self.share_name,
+            file_path=file_name,
+            credential=storage_account_key)
+
+        await file_client.create_file(2048)
+        share_client = self.fsc.get_share_client(self.share_name)
+        snapshot1 = await share_client.create_snapshot()
+
+        data = self.get_random_bytes(1536)
+        await file_client.upload_range(data, offset=0, length=1536)
+        snapshot2 = await share_client.create_snapshot()
+        await file_client.clear_range(offset=512, length=512)
+
+        ranges1 = await file_client.get_ranges(previous_sharesnapshot=snapshot1)
+        ranges2 = await file_client.get_ranges(previous_sharesnapshot=snapshot2['snapshot'])
+
+        # Assert
+        self.assertIsNotNone(ranges1)
+        self.assertIsInstance(ranges1, list)
+        self.assertEqual(len(ranges1), 3)
+        self.assertEqual(ranges1[0]['start'], 0)
+        self.assertEqual(ranges1[0]['end'], 511)
+        self.assertEqual(ranges1[1]['start'], 512)
```

Comment:
512~1023 are cleared range, and the service response body is like this:
```
<Ranges> 

    <Range> 

        <Start>Start Byte</Start> 

        <End>End Byte</Start> 

    </Range> 

    <ClearRange> 

        <Start>Start Byte</Start> 

        <End>End Byte</Start> 

    </ClearRange> 

    <Range> 

        <Start>Start Byte</Start> 

        <End>End Byte</Start> 

    </Range> 

</Ranges> 
```
how do we reflect cleared range in SDK?

Reply:
Offline: had a discussion that currently there's a mismatch between the page blob api and the fileshare api in getting ranges. Currently we are not deseriaalizaing/parsing the `ClearRanges` for fileshares. This is because the pre-existing api does not account for a diff get_range (with a prevsnapshot) and hence, there is no implementation to handle the `ClearRange` that is implemented corretly in page blobs.